go/multipooler/manager/pg_replication.go

```diff
@@ -71,129 +71,144 @@ func formatStandbyList(standbyIDs []*clustermetadatapb.ID) string {
 	}
+	var exists bool
+	if err := executor.ScanSingleRow(result, &exists); err != nil {
+		return false, mterrors.Wrap(err, "failed to scan schema exists result")
+	}
 	return exists, nil
 }
 
 // checkLSNReached checks if the standby has replayed up to or past the target LSN
 func (pm *MultiPoolerManager) checkLSNReached(ctx context.Context, targetLsn string) (bool, error) {
 	queryCtx, cancel := context.WithTimeout(ctx, 500*time.Millisecond)
 	defer cancel()
-
-	var reachedTarget bool
-	err := pm.db.QueryRowContext(queryCtx, "SELECT pg_last_wal_replay_lsn() >= $1::pg_lsn", targetLsn).Scan(&reachedTarget)
+	sql := fmt.Sprintf("SELECT pg_last_wal_replay_lsn() >= %s::pg_lsn", ast.QuoteStringLiteral(targetLsn))
+	result, err := pm.query(queryCtx, sql)
 	if err != nil {
 		return false, mterrors.Wrap(err, "failed to check if replay LSN reached target")
 	}
+	var reachedTarget bool
+	if err := executor.ScanSingleRow(result, &reachedTarget); err != nil {
+		return false, mterrors.Wrap(err, "failed to scan LSN comparison result")
+	}
 	return reachedTarget, nil
 }
 
 // queryReplicationStatus queries PostgreSQL for all replication status fields.
 // This method handles NULL values properly for LSN fields that may be NULL
 // when not in recovery mode or when no WAL has been received/replayed.
 func (pm *MultiPoolerManager) queryReplicationStatus(ctx context.Context) (*multipoolermanagerdatapb.StandbyReplicationStatus, error) {
-	var replayLsn sql.NullString
-	var receiveLsn sql.NullString
-	var isPaused bool
-	var pauseState string
-	var lastXactTime sql.NullString
-	var primaryConnInfo string
-
-	query := `SELECT
-		pg_last_wal_replay_lsn(),
-		pg_last_wal_receive_lsn(),
+	sql := `SELECT
+		pg_last_wal_replay_lsn()::text,
+		pg_last_wal_receive_lsn()::text,
 		pg_is_wal_replay_paused(),
 		pg_get_wal_replay_pause_state(),
-		pg_last_xact_replay_timestamp(),
+		pg_last_xact_replay_timestamp()::text,
 		current_setting('primary_conninfo')`
 
 	queryCtx, cancel := context.WithTimeout(ctx, 500*time.Millisecond)
 	defer cancel()
-
-	err := pm.db.QueryRowContext(queryCtx, query).Scan(
-		&replayLsn,
-		&receiveLsn,
-		&isPaused,
-		&pauseState,
-		&lastXactTime,
-		&primaryConnInfo,
-	)
+	result, err := pm.query(queryCtx, sql)
 	if err != nil {
 		return nil, mterrors.Wrap(err, "failed to query replication status")
 	}
 
-	status := &multipoolermanagerdatapb.StandbyReplicationStatus{
-		IsWalReplayPaused:   isPaused,
-		WalReplayPauseState: pauseState,
+	if result == nil || len(result.Rows) == 0 {
+		return nil, mterrors.Wrap(fmt.Errorf("no rows returned"), "failed to query replication status")
 	}
 
-	if replayLsn.Valid {
-		status.LastReplayLsn = replayLsn.String
+	row := result.Rows[0]
+	status := &multipoolermanagerdatapb.StandbyReplicationStatus{}
+
+	// Get replay LSN (may be NULL)
+	if replayLsn, err := executor.GetString(row, 0); err == nil && replayLsn != "" {
+		status.LastReplayLsn = replayLsn
```

Comment:
Checking for non-empty string changes the meaning of this check. Previously we were checking for NULL via `.Valid`
```
type NullString struct {
	String string
	Valid  bool // Valid is true if String is not NULL
}
```